nav2_costmap_2d/src/costmap_subscriber.cpp

```diff
@@ -21,35 +21,25 @@ namespace nav2_costmap_2d
 {
 
 CostmapSubscriber::CostmapSubscriber(
-  nav2_util::LifecycleNode::SharedPtr node,
+  const nav2_util::LifecycleNode::WeakPtr & parent,
   const std::string & topic_name)
-: CostmapSubscriber(node->get_node_base_interface(),
-    node->get_node_topics_interface(),
-    node->get_node_logging_interface(),
-    topic_name)
-{}
-
-CostmapSubscriber::CostmapSubscriber(
-  rclcpp::Node::SharedPtr node,
-  const std::string & topic_name)
-: CostmapSubscriber(node->get_node_base_interface(),
-    node->get_node_topics_interface(),
-    node->get_node_logging_interface(),
-    topic_name)
-{}
+: topic_name_(topic_name)
+{
+  auto node = parent.lock();
+  costmap_sub_ = node->create_subscription<nav2_msgs::msg::Costmap>(
+    topic_name_,
+    rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable(),
+    std::bind(&CostmapSubscriber::costmapCallback, this, std::placeholders::_1));
+}
 
 CostmapSubscriber::CostmapSubscriber(
-  const rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base,
```

Comment:
This constructor might still be useful

Reply:
As well with footprint subscriber below

Reply:
Most of these objects aren't using the node interfaces anymore. Even the constructor only uses `NodeTopicsInterface`. Are you sure keeping it is a good idea? If we still want to keep a constructor which takes node interfaces, it would be better to remove the unused interfaces right? Like, in this case we can provide a constructor with only a `NodeTopicsInterface` argument.

Reply:
Fair enough. Walking through what I'd do with it I don't think is possible anymore due to needing to  lock